SampleOfferWithoutReward.sol

```diff
@@ -131,6 +131,10 @@ contract SampleOfferWithoutReward {
     // Executing this function before the Offer is signed off by the Client
     // makes no sense as this contract has no money.
     function getDailyPayment() noEther {
+        if (msg.sender != contractor && msg.sender != address(client)) {
```

Comment:
Exactly! Good point.

But still I would like to verify our understanding of who is `msg.sender` in such a case as Christoph pointed.

For the moment we are testing the changes introduced by your Previous PR.